src/custom/state/fee/updater.ts

```diff
@@ -27,27 +42,35 @@ export default function FeesUpdater(): null {
   const now = new Date().toISOString()
 
   useEffect(() => {
-    if (!stateFeesMap || !chainId || !currencyId || !windowVisible) return
+    if (!stateFeesMap || !chainId || !sellToken || !buyToken || !typedValue || !windowVisible) return
+
     registerOnWindow({ addFee })
 
+    const kind = independentField === Field.INPUT ? 'sell' : 'buy'
+    const amount = tryParseAmount(typedValue, (kind === 'sell' ? sellCurrency : buyCurrency) ?? undefined)
+
+    // type check...
+    if (!amount) return
+
     async function runFeeHook({
       feesMap,
       chainId,
-      sellToken
-    }: {
-      feesMap: Partial<FeesMap>
-      chainId: ChainId
-      sellToken: string
-    }) {
+      sellToken,
+      buyToken,
+      amount,
+      kind
+    }: FeeQuoteParams & { feesMap: Partial<FeesMap> }) {
       const currentFee = feesMap[sellToken]?.fee
       const isFeeDateValid = currentFee && isDateLater(currentFee.expirationDate, now)
 
-      if (!isFeeDateValid || !currentFee) {
-        const fee = await getFeeQuote(chainId, sellToken).catch(err => {
+      if (typedValueRef.current !== typedValue || !isFeeDateValid || !currentFee) {
```

Comment:
use a `ref` and state comparison here to make sure to recalc _only_ on amount change from last input